src/lib/secrets/secret_box.mli

```diff
@@ -24,4 +26,4 @@ end
 val encrypt : password:Bytes.t -> plaintext:Bytes.t -> t
 
 (** Decrypt some bytes with a password *)
-val decrypt : password:Bytes.t -> t -> Bytes.t Core.Or_error.t
+val decrypt : password:Bytes.t -> t -> (Bytes.t, Privkey_error.t) Result.t
```

Comment:
Typically it's nice to put errors behind a polymorphic variant so they compose better: see https://keleshev.com/composable-error-handling-in-ocaml